File: 프로그래머스/1/176963. 추억 점수/추억 점수.js
function solution(name, yearning, photo) {


  var answer = [];

  for (let i = 0; i < photo.length; i++) {
    let result = 0;
    for (let j = 0; j < name.length; j++) {
      if (photo[i].includes(name[j])) {
        result += yearning[j];
      }
    }
    answer.push(result);
  }

  console.log(answer);
  return answer;
};

solution(
  ["kali", "mari", "don"],
  [11, 1, 55]	,
  [["kali", "mari", "don"], ["pony", "tom", "teddy"], ["con", "mona", "don"]]
);

File: 프로그래머스/1/176963. 추억 점수/추억 점수.py
from typing import Dict, List

def solution(name: List[str], yearning: List[int], photo: List[List[str]]) -> List[int]:
    scores: Dict[str, int] = dict(zip(name, yearning))
    result = []
    for group in photo:
        total = sum(scores.get(person, 0) for person in group)
        result.append(total)
    return result
